src/main.rs

```diff
@@ -212,7 +235,85 @@ async fn run(cli: Cli) -> anyhow::Result<()> {
     .await?;
 
     log::info!("Done");
-    Ok(())
+    Ok(exit_code)
+}
+
+async fn run_test(test_args: TestArgs) -> anyhow::Result<i32> {
+    let TestArgs {
+        ref command,
+        upload_args,
+    } = test_args;
+    let UploadArgs {
+        ref junit_paths,
+        ref org_url_slug,
+        ref token,
+        ref repo_root,
+        ref repo_url,
+        ref repo_head_sha,
+        ref repo_head_branch,
+        ref repo_head_commit_epoch,
+        tags: _,
+        print_files: _,
+        dry_run: _,
+    } = upload_args;
```

Comment:
If you wan tot avoid adding `ref` you can also take `upload_args` by reference, e.g. `} = &upload_args`. I don't think it matters though.

Reply:
Perfect, thanks!